components/core/src/metrics.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::thread;
+use std::sync::mpsc::{channel, sync_channel, Sender, Receiver, SyncSender};
+use std::sync::{Once, ONCE_INIT};
+use statsd::Client;
+use env;
+
+// Statsd Application name
+pub const APP_NAME: &'static str = "bldr";
```

Comment:
Is this infra meant for Builder specifically or Habitat in general? If it's the former then a common `builder-*` crate might be a better home (`core` generally is a dependency of all Habitat higher level components) or if it's the latter then we may eventually want this as a parameter to using this module.